### PR TITLE
fix: hasError not set true

### PR DESCRIPTION
**Pull request by makeavish**
for server kind spans with 4xx errors

fixes(https://github.com/SigNoz/signoz-otel-collector/issues/147)

### CODE REVIEW

File: exporter/clickhousetracesexporter/clickhouse_exporter.go [hunk 1]
```diff
@@ -157,7 +157,7 @@ func populateOtherDimensions(attributes pcommon.Map, span *Span) {
 			if err == nil && statusString != 0 {
 				statusInt = int64(statusString)
 			}
-			if statusInt >= 400 {
+			if statusInt >= 500 || (statusInt >= 400 && span.Kind != 2) {
```
Comment (srikanthccv) [inline exporter/clickhousetracesexporter/clickhouse_exporter.go:160]: There is a gRPC as well. Please update that.

https://github.com/SigNoz/signoz-otel-collector/blob/8decaf03afdfcd7d7167738642972f2f2afc7f45/exporter/clickhousetracesexporter/clickhouse_exporter.go#L198-L207

Comment (makeavish) [reply exporter/clickhousetracesexporter/clickhouse_exporter.go:160]: For gRPC apparently it's not very clear which are server side errors.
Go through these docs, you will see that most status codes are shared among client and server: https://grpc.github.io/grpc/core/md_doc_statuscodes.html

**Comment by makeavish** [comment]
> As discussed on 26th Jun, Please rely on the span status code for hasError.

@srikanthccv Updated code to only rely on span status code for has error. [Here](https://github.com/SigNoz/signoz-otel-collector/blob/926fba7740d54250942ad39dd289b56cd27b8985/exporter/clickhousetracesexporter/clickhouse_exporter.go#L367) we are checking if span status code is 2 to set hasError to true

**Comment by srikanthccv** [comment]
Can you use the symbol `ptrace.StatusCodeError` from pdata instead of 2 in the linked code?

**Comment by makeavish** [comment]
> Can you use the symbol `ptrace.StatusCodeError` from pdata instead of 2 in the linked code?

Updated

**Review by srikanthccv — APPROVED** [review]

**Pull request merged**